source/ptreferential/tests/ptret_test.cpp

```diff
@@ -428,3 +429,87 @@ BOOST_AUTO_TEST_CASE(find_path_test){
     // Type qui n'existe pas dans le graph : il n'y a pas de chemin
     BOOST_CHECK_THROW(find_path(navitia::type::Type_e::Unknown), ptref_error);
 }
+
+//helper to get default values
+std::vector<nt::idx_t> query(nt::Type_e requested_type, std::string request,
+                             const nt::Data& data,
+                             boost::optional<boost::posix_time::ptime> since = boost::none,
+                             boost::optional<boost::posix_time::ptime> until = boost::none,
+                             bool fail = false) {
+    try {
+       return navitia::ptref::make_query(requested_type, request, {}, navitia::type::OdtLevel_e::all, since, until, data);
+    } catch (const navitia::ptref::ptref_error& e) {
+        if (fail) {
+            throw e;
+        }
+        BOOST_CHECK_MESSAGE(false, " error on ptref request" + std::string(e.what()));
+        return {};
+    }
+}
+
+/*
+ * Test the filtering on the period
```

Comment:
Maybe add a little ascii art on departure times (which is the considered value for filtering) like : 
Date          12        13        14        15
A                 -      08:00    08:00       -
B            10:00        -       10:00   10:00
C            10:00        -           -          -

Not sure I got it all right, though.